### PR TITLE
feat(auth): enable nodemailer SES transport

### PR DESCRIPTION
**Pull request by dannycoates**
## Because

- I'd like to deprecate email-service

## This pull request

- Adds the nodemailer SES transport
- The AWS config is set by the environment and is used when no SMTP user:password are set
- We _could_ use liveConfig to run a subset of addresses through direct SES vs email-service
  - https://github.com/mozilla/fxa/blob/auth-ses/packages/fxa-auth-server/lib/senders/select_email_services.js#L18
- Gets used 100% when email-service has been configed off with `EMAIL_CONFIG_USE_REDIS=false`
- Resurrects some old code from https://github.com/mozilla/fxa-auth-server/pull/1934 🧟 🧙

## Notes

After this we'll have 3 ways to send via SES: API direct, SMTP, fxa-email-service. The goal is to use API direct and eliminate fxa-email-service, with an SMTP "backup" in case we need to support other providers.

## Issue that this pull request solves

Closes: #9725

### CODE REVIEW

File: _dev/pm2/infrastructure.config.js
Comment (dannycoates) [inline _dev/pm2/infrastructure.config.js:71]: removing email-service from local dev

File: packages/fxa-auth-server/bin/key_server.js
Comment (dannycoates) [inline packages/fxa-auth-server/bin/key_server.js:116]: `error` was no longer used downstream so I reused it's slot for `bounces`

File: packages/fxa-auth-server/config/index.ts
Comment (dannycoates) [inline packages/fxa-auth-server/config/index.ts:392]: resurrecting the pre-email-service config from https://github.com/mozilla/fxa-auth-server/pull/1934

File: packages/fxa-auth-server/config/index.ts
Comment (dannycoates) [inline packages/fxa-auth-server/config/index.ts:404]: changed this from 1 year to a more sane 7 days

File: packages/fxa-auth-server/config/index.ts
Comment (dannycoates) [inline packages/fxa-auth-server/config/index.ts:414]: used to be 1 year. no wonder why people have problems with bounces 😳

File: packages/fxa-auth-server/lib/bounces.js
Comment (dannycoates) [inline packages/fxa-auth-server/lib/bounces.js:1]: All from https://github.com/mozilla/fxa-auth-server/pull/1934

File: packages/fxa-auth-server/lib/senders/email.js
Comment (dannycoates) [inline packages/fxa-auth-server/lib/senders/email.js:512]: we check bounces 🏀

File: packages/fxa-auth-server/test/local/bounces.js
Comment (dannycoates) [inline packages/fxa-auth-server/test/local/bounces.js:1]: used unit tests - good condition, runs, new promises, $400 OBO

original owner: https://github.com/mozilla/fxa-auth-server/pull/1934

File: packages/fxa-auth-server/test/local/senders/email.js
Comment (dannycoates) [inline packages/fxa-auth-server/test/local/senders/email.js:1709]: nothing was using the result so i changed the `send` signature to return void

File: packages/fxa-auth-server/lib/senders/email.js
Comment (dannycoates) [inline packages/fxa-auth-server/lib/senders/email.js:522]: This is copied from below in the `sendMail` callback, and is related to @biancadanforth's PR #10543 ... but I think it'd be better as it's own distinct log entry. I'm noting it here so that if we do change it we'll know about it for monitoring.

File: packages/fxa-auth-server/config/index.ts
Comment (biancadanforth) [reply packages/fxa-auth-server/config/index.ts:414]: Yeah I [recommended changing these values](https://docs.google.com/document/d/1lXa6ZwxtyXbzB-YQOr0HuRRwtKzqE2etiWUwEN9IRok/edit?disco=AAAAPRMDFnc) as a short-term fix when I was poking around for #10527 .

File: packages/fxa-auth-server/lib/senders/email.js
Comment (biancadanforth) [reply packages/fxa-auth-server/lib/senders/email.js:522]: What do you mean by making it its own distinct log entry?

**Review by vbudhram — APPROVED** [review]
@dannycoates I gave this a test locally and found that this line needs to pass `row` https://github.com/mozilla/fxa/blob/a561ae1f32344a93c4d6ef68a1427fa472a89b08/packages/fxa-shared/db/models/auth/email-bounce.ts#L62. After that it worked like a charm with manually entered bounces.

Awesome work!

File: packages/fxa-auth-server/config/index.ts
Comment (biancadanforth) [reply packages/fxa-auth-server/config/index.ts:414]: I didn't have a specific suggestion, as I don't understand SES' deliverability score too well and how lowering the threshold (especially substantially) might affect it. If you do lower the threshold, I would let Janet/Anna/Chelsea know, as it may substantively impact (hopefully improve?) our bounce rates or other email delivery metrics.

File: packages/fxa-auth-server/lib/senders/email.js
Comment (dannycoates) [reply packages/fxa-auth-server/lib/senders/email.js:522]: as in with it's own "name" that we can query like `email.bounce` instead of `mailer.send.error`

**Comment by biancadanforth** [comment]
@dannycoates Are you going to merge this today before we tag? If so, I'll wait for it to merge, so I can rebase my PR off of it for the `template` property in the `log.error`.

**Comment by dannycoates** [comment]
> @dannycoates Are you going to merge this today before we tag? If so, I'll wait for it to merge, so I can rebase my PR off of it for the `template` property in the `log.error`.

I'd like to but go ahead and merge yours first

**Pull request merged**
